pkg/asset/agent/image/ignition.go

```diff
@@ -127,25 +125,14 @@ func (a *Ignition) Generate(dependencies asset.Parents) error {
 		return err
 	}
 
-	agentMirror := &mirror.AgentMirror{}
-	dependencies.Get(agentMirror)
-
-	// Mount files for assisted-service
-	mirrorRegistriesMount := ""
-	caBundleMount := ""
-	for _, file := range agentMirror.FileList {
-		if file.Filename == mirror.RegistriesConfFilename {
-			mirrorRegistriesMount = fmt.Sprintf("-v %s:/etc/containers/registries.conf:z", filepath.Join("/etc/assisted", file.Filename))
-		}
-		if file.Filename == mirror.CaBundleFilename {
-			caBundleMount = fmt.Sprintf("-v %s:/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem:z", filepath.Join("/etc/assisted", file.Filename))
-		}
-	}
+	registriesConfig := &mirror.RegistriesConf{}
+	registryCABundle := &mirror.CaBundle{}
+	dependencies.Get(registriesConfig, registryCABundle)
 
 	// Get the mirror for release image
 	releaseImageMirror := ""
 	source := strings.Split(agentManifests.ClusterImageSet.Spec.ReleaseImage, ":")
-	for _, config := range agentMirror.MirrorConfig {
+	for _, config := range registriesConfig.MirrorConfig {
 		if config.Location == source[0] {
```

Comment:
releaseImageMirror is left as empty string because a match is never found. When dev-scripts is configured with MIRROR_IMAGES=true I see the following values for config.Location and source[0] in the for loop:
```
level=info msg=RWSU config.Location: registry.ci.openshift.org/ocp/release
level=info msg=RWSU source[0]: virthost.ostest.test.metalkube.org
level=info msg=RWSU config.Location: quay.io/openshift-release-dev/ocp-v4.0-art-dev
level=info msg=RWSU source[0]: virthost.ostest.test.metalkube.org
```
agentManifests.ClusterImageSet.Spec.ReleaseImage=virthost.ostest.test.metalkube.org:5000/localimages/local-release-image:4.11.0-0.nightly-2022-07-11-080250

Reply:
@bfournie could this be a limitation/problem of the dev-scripts config? I'd have expected a mirror to match here, and that some stuff in assisted-service might not work if none does.

Reply:
assisted-service should handle it fine, and does in my testing - it checks the mirror first and uses the releaseImage if not set. But we could set the releaseImageMirror [here](https://github.com/openshift/installer/blob/89ebbd566cb7d7479e828f87f0248915f0eea946/pkg/asset/agent/image/ignition.go#L149)
if its also in the mirror section of the config, then the conditional would work OK

Reply:
Also, using the release image set to the mirror is common setting for users, see for example Eran's comment #2 [here](https://issues.redhat.com/browse/MGMT-10209)

Reply:
OK, I realised I have to bite the bullet and pass another boolean in the config. I tested that both with and without MIRROR_IMAGES enabled in dev-scripts and it now works. Thanks everybody!